protocol/protocol.tex

```diff
@@ -7643,92 +8675,174 @@
 \end{formulae}
 \vspace{-1ex}
 which is equivalent to:
 \begin{formulae}
-  \item $\ValueCommit{\ValueCommitRand}(\Value) := \scalarmult{\Value}{\ValueCommitValueBase}
-                                                 + \scalarmult{\ValueCommitRand}{\ValueCommitRandBase}$.
+  \item $\ValueCommit{Orchard}{\ValueCommitRand}(\Value) := \scalarmult{\Value}{\ValueCommitValueBase{Orchard}}
+                                                          + \scalarmult{\ValueCommitRand}{\ValueCommitRandBase{Orchard}}$.
 \end{formulae}
+} %orchard
 
 \begin{securityrequirements}
   \item $\HomomorphicPedersenCommitAlg$ must be a computationally binding and at least
-        computationally hiding \commitmentScheme, for a given personalization input $D$.
-  \item $\ValueCommitAlg$ must be a computationally binding and at least
-        computationally hiding \commitmentScheme.
+        computationally hiding \commitmentScheme\orchard{ on both curves}, for a given personalization
+        input $D$.
+  \item $\ValueCommitAlg{Sapling}$\orchard{ and $\ValueCommitAlg{Orchard}$} must be\notorchard{ a} computationally
+        binding and at least computationally hiding \notorchard{\commitmentScheme}\notbeforeorchard{\commitmentSchemes}.
 \end{securityrequirements}
 
 (They are in fact unconditionally hiding \commitmentSchemes.)
-}
+} %sapling
+
+
+\orchard{
+\introsection
+\lsubsubsubsection{Sinsemilla commitments}{concretesinsemillacommit}
+
+\crossref{concretesinsemillahash} defines a \xSinsemillaHash construction.
+We construct \defining{\xSinsemillaCommitments} by reusing that construction,
+and adding a randomized point on the \pallasCurve (see \crossref{pallasandvesta}):
+
+\begin{formulae}
+  \item $\SinsemillaCommit{r}(D, M) :=
+           \SinsemillaHashToPoint(D \bconcat \ascii{-M}, M) + \scalarmult{r}{\GroupPHash\Of{D \bconcat \ascii{-r}, \ascii{}}}$
+\end{formulae}
+
+See \todo{...} for rationale and efficient circuit implementation of this function.
+
+The commitment scheme $\NoteCommitAlg{Orchard}$ specified in \crossref{abstractcommit} is
+instantiated as follows using $\SinsemillaCommitAlg$:
+
+\begin{formulae}
+  \item $\NoteCommit{Orchard}{\NoteCommitRand}(\DiversifiedTransmitBaseRepr, \DiversifiedTransmitPublicRepr, \Value) :=
+           \SinsemillaCommit{\NoteCommitRand}\left(\ascii{z.cash:Orchard-NoteCommit},
+             \DiversifiedTransmitBaseRepr \bconcat \DiversifiedTransmitPublicRepr \bconcat \ItoLEBSPOf{64}{\Value}\right)$
+  \item $\NoteCommitGenTrapdoor{Orchard}()$ generates the uniform distribution on $\GF{\ParamP{r}}$.
+\end{formulae}
+
+The commitment scheme $\CommitIvkAlg$ specified in \crossref{abstractcommit} is
+instantiated as follows using $\SinsemillaCommitAlg$:
+
+\begin{formulae}
+  \item $\CommitIvk{\CommitIvkRand}(\AuthSignPublicX, \NullifierKey) :=
+           \SinsemillaCommit{\NoteCommitRand}\left(\ascii{z.cash:Orchard-CommitIvk},
+             \AuthSignPublicXRepr \bconcat \NullifierKeyRepr\right)$
```

Comment:
Type error: `SinsemillaCommit` returns `P`, but `CommitIvk` needs to return `F_P`. This should be `SinsemillaShortCommit` (which is currently undefined).